### PR TITLE
Combobox vNext styles

### PR DESCRIPTION
**Pull request by smhigley**
Part of the Combobox convergence epic (#18133), adds styles for the combobox button & listbox/options dropdown.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c1502884f4f422e6e7c3beba1cd07a2318efd37e'>c1502884f4f422e6e7c3beba1cd07a2318efd37e</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts [hunk 1]
```diff
@@ -1,70 +1,201 @@
...
-      borderBottomWidth: '2px',
     },
-  },
 
-  expandIcon: {
-    flexGrow: 0,
-    flexShrink: 0,
-    ...shorthands.padding('4px'),
+    // bottom focus border, shared with Input, Select, and SpinButton
+    ':after': {
```
Comment (spmonahan) [inline packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:63]: nit: prefer `::after` to `:after`.

See: https://developer.mozilla.org/en-US/docs/Web/CSS/::after#syntax

File: packages/react-combobox/src/components/OptionGroup/useOptionGroupStyles.ts [hunk 2]
```diff
@@ -1,25 +1,49 @@
...
+      paddingBottom: horizontalSpacing.xs,
+      marginBottom: horizontalSpacing.xs,
+    },
   },
 
-  label: {},
+  label: {
+    ...shorthands.borderRadius(tokens.borderRadiusMedium),
+    color: tokens.colorNeutralForeground3,
+    columnGap: horizontalSpacing.xxs,
```
Comment (spmonahan) [inline packages/react-combobox/src/components/OptionGroup/useOptionGroupStyles.ts:40]: Is this property necessary? I don't think it has any effect here.

Comment (smhigley) [reply packages/react-combobox/src/components/OptionGroup/useOptionGroupStyles.ts:40]: Good point, I can remove it. It was specified in the design spec, but I don't think there's any actual reason it's needed.

Comment (khmakoto) [reply packages/react-combobox/src/components/OptionGroup/useOptionGroupStyles.ts:40]: Yeah, since this is `display: 'block'` we should probably remove it.

File: packages/react-combobox/src/components/Option/useOption.tsx [hunk 3]
```diff
@@ -73,6 +81,7 @@ export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElemen
     root: getNativeElementProps('div', {
       ref,
       role: 'option',
+      'aria-disabled': disabled ? 'true' : null,
```
Comment (khmakoto) [inline packages/react-combobox/src/components/Option/useOption.tsx:84]: Should this be `undefined` instead of `null` for the false case?

**Review by khmakoto — APPROVED** [review]
Approved with very minor comments 😄

**Review by spmonahan — APPROVED** [review]

File: packages/react-combobox/src/components/Option/useOption.tsx (hunk 3 above)
Comment (spmonahan) [reply packages/react-combobox/src/components/Option/useOption.tsx]: FWIW, I agree that `undefined` is more idiomatic React but when I tested it `null` gave the same result as `undefined`.

File: packages/react-combobox/src/components/ComboButton/ComboButton.test.tsx [hunk 4]
```diff
@@ -8,6 +8,8 @@ describe('ComboButton', () => {
     Component: ComboButton,
     displayName: 'ComboButton',
     primarySlot: 'content',
+    // don't test deprecated className export on new components
+    disabledTests: ['component-has-static-classname-exported'],
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/ComboButton/ComboButton.test.tsx:12]: @TristanWatanabe We should bring up removing this test at the next v-build team meeting.

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts [hunk 5]
```diff
@@ -1,70 +1,201 @@
...
     },
-  },
 
-  expandIcon: {
-    flexGrow: 0,
-    flexShrink: 0,
-    ...shorthands.padding('4px'),
+    // bottom focus border, shared with Input, Select, and SpinButton
+    '::after': {
+      boxSizing: 'border-box',
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:64]: Nit: With absolute positioning, do you need boxSizing set?

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts [hunk 6]
```diff
@@ -1,70 +1,201 @@
...
+      borderBottomLeftRadius: tokens.borderRadiusMedium,
+      borderBottomRightRadius: tokens.borderRadiusMedium,
+      ...shorthands.borderBottom('2px', 'solid', tokens.colorCompoundBrandStroke),
+      clipPath: 'inset(calc(100% - 2px) 0 0 0)',
+      transform: 'scaleX(0)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.ultraFast,
+      transitionDelay: motionCurves.accelerateMid,
+    },
+    ':focus-within:after': {
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:80]: Nit:   Should this be ::after ?

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts [hunk 7]
```diff
@@ -1,70 +1,201 @@
...
+      transform: 'scaleX(1)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.normal,
+      transitionDelay: motionCurves.decelerateMid,
+    },
+    ':focus-within:active:after': {
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:86]: Same comment about ::after

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts [hunk 8]
```diff
@@ -1,70 +1,201 @@
...
+      borderBottomColor: tokens.colorCompoundBrandStrokePressed,
+    },
   },
 
   content: {
-    flexGrow: 1,
-    backgroundColor: 'transparent',
-    ...shorthands.border('none'),
+    alignItems: 'center',
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:92]: Nit: Consider also setting alignContent.  My understanding is there are cases where alignContent is meaningful when the items don't take up all the available space.

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts [hunk 9]
```diff
@@ -1,70 +1,201 @@
...
+    lineHeight: tokens.lineHeightBase400,
+    ...shorthands.padding('7px', horizontalSpacing.m),
+  },
+
+  // appearance variants
+  outline: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border('1px', 'solid', tokens.colorNeutralStroke1),
+    borderBottomColor: tokens.colorNeutralStrokeAccessible,
+    color: 'red',
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:130]: Did you mean to leave color: 'red' in here or what that from debugging?

File: packages/react-combobox/src/components/Listbox/useListboxStyles.ts [hunk 10]
```diff
@@ -1,24 +1,35 @@
...
 
-/**
- * @deprecated Use `listboxClassNames.root` instead.
- */
-export const listboxClassName = 'fui-Listbox';
 export const listboxClassNames: SlotClassNames<ListboxSlots> = {
   root: 'fui-Listbox',
 };
 
+// TODO(sharing) use theme values once available
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/Listbox/useListboxStyles.ts:10]: Nit: You could put these constants in a shared place within this package to await being shared across fluent.

File: packages/react-combobox/src/components/Option/useOption.tsx [hunk 11]
```diff
@@ -48,6 +50,12 @@ export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElemen
   const selected = key ? !!selectedOptions.find(option => option.key === key) : false;
   const optionValue = getValueString(value, props.children);
 
+  // check icon
+  let CheckIcon = <CheckmarkFilled />;
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/Option/useOption.tsx:54]: Consider if React.memo would be good to use here.

File: packages/react-combobox/src/components/Option/useOptionStyles.ts [hunk 12]
```diff
@@ -1,63 +1,127 @@
...
   },
 
-  // these are testing-only styles
   active: {
-    outlineWidth: '2px',
-    outlineStyle: 'solid',
-    outlineColor: 'black',
+    // taken from @fluentui/react-tabster
+    // cannot use createFocusIndicatorStyle() directly, since we aren't using the :focus selector
+    ':after': {
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/components/Option/useOptionStyles.ts:45]: Same comment about ::after

File: packages/react-combobox/src/stories/ComboboxDefault.stories.tsx [hunk 13]
```diff
@@ -10,7 +10,9 @@ export const Default = (props: Partial<ComboboxProps>) => {
       <label id={comboId}>Best pet</label>
       <Combobox aria-labelledby={comboId} placeholder="Select an animal" {...props}>
         {options.map(option => (
-          <Option key={option}>{option}</Option>
+          <Option key={option} disabled={option === 'Ferret'}>
```
Comment (GeoffCoxMSFT) [inline packages/react-combobox/src/stories/ComboboxDefault.stories.tsx:13]: Why should the Ferret be disabled? The hamster in the ball would be my choice. ;-)

**Review by GeoffCoxMSFT — APPROVED** [review]

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts (hunk 5 above)
Comment (smhigley) [reply packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:64]: it's because of the height + border, the absolute positioning doesn't affect the internal sizing.

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts (hunk 9 above)
Comment (smhigley) [reply packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:130]: definitely just a debugging artifact 🤣

File: packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts (hunk 8 above)
Comment (smhigley) [reply packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts:92]: `alignContent` affects multi-line flex alignment, which isn't allowed in the combobox trigger/faceplate. I added an explicit `flexWrap: nowrap` to the styles to make that clearer 👍

File: packages/react-combobox/src/components/Option/useOption.tsx (hunk 11 above)
Comment (smhigley) [reply packages/react-combobox/src/components/Option/useOption.tsx:54]: I don't think we memoize icons elsewhere right now, so I think I'll leave as-is for now. It'd be an interesting sync question though :D

Comment (spmonahan) [reply packages/react-combobox/src/components/Option/useOption.tsx:54]: I know @tomi-msft is looking at icon rendering performance. He may have some guidance here.
